app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,13 @@
+class CategoriesController < ApplicationController
+  
+  def index
+    @categories = Category.all
+  end
+
+  def show
+    @categories = Category.all
+    @category = Category.find(params[:id])
+    @products = @category.products
+  end
+
+end
```

Comment:
You don't need this controller

Reply:
You mean the index method? Because the show method for categories we are using in order to display products on each category page (show view). If we take that away then the test fails since we can't visit the show page for Categories? So we think we still need the actual CategoriesController, and the show method with those instance variables, but can take away the index method. Makes sense?